pkg/frontend/nvme_controller.go

```diff
@@ -75,15 +74,25 @@ func NewNvmeTCPTransport(listenAddr string) NvmeTransport {
 	}
 }
 
-func (c *nvmeTCPTransport) Params(_ *pb.NvmeController, nqn string) (spdk.NvmfSubsystemAddListenerParams, error) {
+func (c *nvmeTCPTransport) Params(ctrl *pb.NvmeController, nqn string) (spdk.NvmfSubsystemAddListenerParams, error) {
 	result := spdk.NvmfSubsystemAddListenerParams{}
 	result.Nqn = nqn
-	result.SecureChannel = false
+
+	var port int32
+	if *ctrl.Spec.NvmeControllerId < 0 {
+		result.SecureChannel = true
+		port = -*ctrl.Spec.NvmeControllerId
+	} else {
+		result.SecureChannel = false
+		port = *ctrl.Spec.NvmeControllerId
+	}
+
 	result.ListenAddress.Trtype = "tcp"
 	result.ListenAddress.Traddr = c.listenAddr.String()
-	result.ListenAddress.Trsvcid = c.listenPort
+	result.ListenAddress.Trsvcid = fmt.Sprintf("%s%d", c.listenPort, port)
 	result.ListenAddress.Adrfam = c.protocol
```

Comment:
Ugly hack! maybe introduce new cli flag for TLS trid ? and use it here instead of attaching ctrl id to port?

Reply:
went in different direction